### ISSUE TITLE
Organization's DQ Rules can be inadvertently deleted

### ISSUE THREAD

**Issue by adrian-lara** [SEED-platform/seed#2202]
### Expected Behavior
When saving a DQ Rule as Valid Data but not applying a Label, an error is shown, and nothing is saved.

### Actual Behavior
The error-causing DQ Rule and all DQ rules listed after it will be deleted.

If the first DQ rule is the reason the error is thrown, all DQ Rules are deleted. This leads to the DQ Rules being completely reset to the default set.

### Steps to Reproduce
1. Try to save a Rule as Valid Data and don't apply a Label.
<img width="1144" alt="Screen Shot 2020-05-08 at 12 03 42 AM" src="https://user-images.githubusercontent.com/30608004/81376245-5c9af080-90c0-11ea-8255-6365a6431f1e.png">
2. See error as expected:
<img width="1144" alt="Screen Shot 2020-05-08 at 12 04 13 AM" src="https://user-images.githubusercontent.com/30608004/81376282-6cb2d000-90c0-11ea-9a9e-8fc96eebc184.png">
3. Refresh page - all rules are reset.
<img width="1144" alt="Screen Shot 2020-05-08 at 12 04 32 AM" src="https://user-images.githubusercontent.com/30608004/81376617-29a52c80-90c1-11ea-915b-5acd8d77d93e.png">

### Instance Information
`develop` `2.7.0` and `2.7.1`

**Comment by adrian-lara** [comment]
@nllong , @axelstudios , @lainsworth8801 Here are my thoughts on this DQ bug. 

**TL;DR**: Long term, no major extra steps needed as this would've been fixed anyway. Short-term, we should prioritize a patch to desired releases with a fix.

### Long term:
Ideally, the long term solution would be to avoid destroying and recreating the Rules whenever "Save Changes" is clicked or, more specifically, whenever the `save_data_quality_rules` endpoint is hit. According to #2179 , we'll be deprecating this single endpoint in favor of individual POST endpoints for a single Rule. This individual POST will be non-destructive and would naturally inherit the "checks" performed by the old `save_data_quality_rules` endpoint.

As discussed separately, the frontend could also be updated to disable the "Save Changes" button when a "check" would eventually fail. This particular interaction needs further design before implementation.

This update would be included in future releases. Previous releases wouldn't need to be patched.

### Short term:
Given its high impact - users can potentially to delete their Rules - in the short-term, I'd recommend the following, [using this as a reference](https://github.com/SEED-platform/seed/blob/3ab755acc706dea1af6e5384fdb2a4f7545ebfc6/seed/views/data_quality.py#L405):
- Move any "checks" that trigger a 400 to before the records are deleted. As of this writing, there is only one check to ensure that a "Valid Data" Rule has a label, but more are expected.
- Update the try-except logic to "...except and continue rebuilding Rules", returning a JSON response at the end that includes a message that Rules were lost/deleted. I loosely imagine that this would be a 200 response with a message, but I could be convinced against that.

Once the "checks" are moved, the bug will be triggered far less often. In the off chance that a Rule creation throws an error, we'll no longer lose all the proceeding Rules.

Ideally, this fix should be implemented in `develop` as well as on any previous releases where the "checks" exist after the Rules are deleted. Before the "checks" existed, it would have still been possible to delete Rules, though far less likely via the frontend. 

As of this writing, the following releases have the more easily triggered version of this bug (/the "checks" version) - I'll leave it up to you all as to whether we end up patching them:
- [2.7.1](https://github.com/SEED-platform/seed/blob/v2.7.1/seed/views/data_quality.py#L405)
- [2.7.0](https://github.com/SEED-platform/seed/blob/v2.7.0/seed/views/data_quality.py#L405)

**Comment by RDmitchell** [comment]
instance: dev1
SHA: fd75dea4
Org: LBNL Test 200

* Set a rule to be Valid Data, without a label
* Program put up error message saying Valid Data must have a label
![image](https://user-images.githubusercontent.com/6314950/88217845-fda33d00-cc13-11ea-88d6-180982a47ca2.png)
* Refreshed the page
* Rules went back to their original state, ie, the label was no longer deleter.
* All the other rules stayed the same. 

Closing
